libfoundation/src/foundation-string.cpp

```diff
@@ -6963,6 +6963,109 @@ bool MCStringGetNumericValue(MCStringRef self, double &r_value)
         return false;
 }
 
+MC_DLLEXPORT bool
+MCStringNormalizeLineEndings(MCStringRef p_input, 
+                             MCStringLineEndingStyle p_to_style, 
+                             bool p_normalize_ps_ls,
+                             MCStringRef& r_output, 
+                             MCStringLineEndingStyle* r_original_style)
+{
+    MCStringLineEndingStyle t_original_style = kMCStringLineEndingStyleLF;
+    uindex_t t_first_lf = 0;
+    uindex_t t_first_cr = 0;
+    
+    if (MCStringFirstIndexOfChar(p_input, '\r', 0, kMCStringOptionCompareExact, t_first_cr))
+    {
+        if (MCStringFirstIndexOfChar(p_input, '\n', 0, kMCStringOptionCompareExact, t_first_lf))
+        {
+            if (t_first_cr + 1 == t_first_lf)
+            {
+                t_original_style = kMCStringLineEndingStyleCRLF;
+            }
+            else if (t_first_cr < t_first_lf)
+            {
+                t_original_style = kMCStringLineEndingStyleCR;
+            }
+        }
+        else
+        {
+            t_original_style = kMCStringLineEndingStyleCR;
+        }
+    }
+    
+    // normalize input to LF line endings first
+    MCAutoStringRef t_mutable_input;
+    if (!MCStringMutableCopy(p_input, &t_mutable_input))
+    {
+        return false;
+    }
+    if (!MCStringFindAndReplace(*t_mutable_input, 
+                                MCSTR("\r\n"), 
+                                MCSTR("\n"), 
+                                kMCStringOptionCompareExact))
+    {
+        return false;
+    }
```

Comment:
Don't think you are missing anything.  Just making sure the original code wasn't there for some special reason.